# In-class_Assignment-2 
Please answer the questions as in the attached document and submit the answer. Please note, you have to submit jupyter notebook file for answering the question using python. Good luck

In [13]:
data = [28,43,58,49,45,55,60,49,51]

In [26]:
N = len(data)
print(f'The length of data is {N}')

The length of data is 9


## a) Compute:

### Range

In [68]:
range = round(max(data) - min(data),2)
print(f'The range is {range}')

The range is 32


### Interquartile Range

In [69]:
sorted_data = sorted(data)
print(f'The sorted data is: {sorted_data}')

def Q(p):
    qpos = (N+1) * p
    print(f'qpos = {qpos}')
    if int(qpos) == qpos:
        return sorted_data[int(qpos)-1]
    return round((sorted_data[int(qpos)-1] + sorted_data[int(qpos)]) / 2, 2)

The sorted data is: [28, 43, 45, 49, 49, 51, 55, 58, 60]


In [70]:
Q1 = Q(0.25)
Q3 = Q(0.75)
print(f'The first quartile is {Q1}')
print(f'The third quartile is {Q3}')
IQR = Q3 - Q1
print(f'The IQR is {IQR}')

qpos = 2.5
qpos = 7.5
The first quartile is 44.0
The third quartile is 56.5
The IQR is 12.5


### Median

In [71]:
Median = Q(0.5)
print(f'The median is: {Median}')

qpos = 5.0
The median is: 49


### Mode

In [72]:
import statistics
Mode = statistics.mode(data)
print(f'The Mode is {Mode}')

The Mode is 49


### First Quartile

In [73]:
print(f'The first quartile is {Q1}')

The first quartile is 44.0


### Third Quartile

In [74]:
print(f'The third quartile is {Q3}')

The third quartile is 56.5


## b) Compute the sample variance and sample standard deviation (Round your answers to two decimal places)

### Mean

In [75]:
Mean = round(sum(data)/N, 2)
print(f'The mean is {Mean}')

The mean is 48.67


### Sample variance

In [76]:
Variance = round(sum([(x - Mean)**2 for x in data]) / N, 2)
print(f'The variance is {Variance}')

The variance is 81.56


### Sample standard deviation

In [77]:
import math
StandardDeviation = round(math.sqrt(Variance), 2)
print(f'The standard deviation is {StandardDeviation}')

The standard deviation is 9.03


## c) A sample of air quality index readings for Anaheim provided a sample mean of 48.5, a sample variance of 136, and a sample standard deviation of 11.66. 
What comparisons can you make between the air quality in Pomona and that in Anaheim on the basis of these descriptive statistics?

The average air quality in Anaheim is **Slighly lower (48.5 vs 48.67) than** the average air quality in Pomona.

The variability is greater in **Anaheim (11.66 vs 9.03)**